pkg/containertools/runner.go

```diff
@@ -22,14 +22,75 @@ type CommandRunner interface {
 type ContainerCommandRunner struct {
 	logger        *logrus.Entry
 	containerTool ContainerTool
+	config        *RunnerConfig
+}
+
+type RunnerConfig struct {
+	SkipTLS bool
+}
+
+type RunnerOption func(config *RunnerConfig)
+
+func SkipTLS(skip *bool) RunnerOption {
+	return func(config *RunnerConfig) {
+		if skip != nil {
+			config.SkipTLS = *skip
+		}
+	}
+}
+
+func (r *RunnerConfig) apply(options []RunnerOption) {
+	for _, option := range options {
+		option(r)
+	}
+}
+
+func (r *ContainerCommandRunner) argsForCmd(cmd string, args... string) []string {
+	cmdArgs := []string{cmd}
+	switch r.containerTool {
+	case PodmanTool:
+		switch cmd {
+		case "build", "pull", "push", "login", "search":
+			// --tls-verify is a valid flag for these podman subcommands
+			if r.config.SkipTLS {
+				cmdArgs = append(cmdArgs, "--tls-verify=false")
+			}
+		}
+	case DockerTool:
+		if !r.config.SkipTLS {
+			cmdArgs = append(cmdArgs, "--tls")
+		}
+	default:
+	}
+	cmdArgs = append(cmdArgs, args...)
+	return cmdArgs
+}
+
+func defaultConfig(toolName string) *RunnerConfig {
+	switch toolName {
+	case "docker":
+		// docker disables tls verify by default, mimic that behavior
+		return &RunnerConfig{
+			SkipTLS: true,
+		}
+	case "podman":
```

Comment:
As @njhale mentioned I'm not sure we want to have different default tls-skip behavior based on the container building tool the user chooses.

Reply:
Modified this to a global default with tls-skip being false.